scripts/Muon/GUI/Common/contexts/fitting_contexts/general_fitting_context.py

```diff
@@ -0,0 +1,90 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import IFunction
+from Muon.GUI.Common.contexts.fitting_contexts.basic_fitting_context import BasicFittingContext
+
+
+class GeneralFittingContext(BasicFittingContext):
+
+    def __init__(self, allow_double_pulse_fitting: bool = False):
```

Comment:
Ditto on this, it can be 100% replaced with a Python dataclass and we can let Python manually generate the properties